airly/__init__.py

```diff
@@ -48,3 +48,13 @@ def create_measurements_session_point(
         return MeasurementsSession(
             self._rh, MeasurementsSession.Mode.POINT,
             latitude=latitude, longitude=longitude)
+
+    @property
+    def requests_remaining(self):
+        """Returns the remaining number of requests."""
```

Comment:
Your change looks great. Could you add some unit tests with sample headers provided?

Reply:
I'm not familiar with `unittest`, so far I have only used `pytest`. Do you have any examples of how to mock http server responses?

Reply:
I set some simple responses in AirlyTestCase.set_up_next_response_from_file metod. I think you could improve this method to accept headers as additional parameter.

General approach is presented here: https://stackoverflow.com/a/28507806

Reply:
`airly._private._RequestsHandler` is patched in every tests and headers logic is in this method so we need new test with patched `aiohttp` response. Am I right?

Reply:
I looked at possibilities and I think we should add unit tests for real, not patched _RequestsHandler. This looks like a very promising way to mock ClientSession provided to _RH constructor: https://github.com/pnuckowski/aioresponses

Reply:
I also found `aioresponses` and I think it might be a good solution. Unfortunately, I have a very busy week so I won't play with it until Saturday or Sunday.

Reply:
Ok with this code I have working headers tests for `pytest`:
```python
"""Tests for airly package."""
import json
from unittest.mock import AsyncMock, Mock

import pytest

from airly import Airly

LATITUDE = 12
LONGITUDE = 21


async def test_valid_headers():
    with open("data/measurements_typical.json") as file:
        data = json.load(file)
    headers = {"X-RateLimit-Limit-day": "1000", "X-RateLimit-Remaining-day": "993"}
    with Mock(
        get=AsyncMock(side_effect=data),
        headers=Mock(side_effect=headers),
        __enter__=Mock(),
        __exit__=Mock(),
    ) as mock_session:
        airly = Airly("abcdef", mock_session)
        measurements = airly.create_measurements_session_point(LATITUDE, LONGITUDE)
        await measurements.update()
        assert airly.requests_per_day == 1000
        assert airly.requests_remaining == 993

async def test_invalid_headers():
    with open("tests/data/measurements_typical.json") as file:
        data = json.load(file)
    headers = {}
    with Mock(
        get=AsyncMock(side_effect=data),
        headers=Mock(side_effect=headers),
        __enter__=Mock(),
        __exit__=Mock(),
    ) as mock_session:
        airly = Airly("abcdef", mock_session)
        measurements = airly.create_measurements_session_point(LATITUDE, LONGITUDE)
        await measurements.update()
        assert airly.requests_per_day == None
        assert airly.requests_remaining == None
```
Now I have to "translate" this to `unittest` ;)

Reply:
I think we got it. Please review.

Reply:
I made a stupid mistake, these tests don't work 🤦‍♂️

Reply:
Now I have used `aioresponses` and the tests are working fine. Sorry for the confusion. Beginner's mistakes 😁